lib/src/flagged/flagged_bloc.dart

```diff
@@ -87,26 +92,33 @@ class FlaggedBloc extends Bloc<FlaggedEvent, FlaggedState> {
   void _registerListeners() {
       if (!_listenersRegistered) {
         _listenersRegistered = true;
-        repositoryStreamHandler = RepositoryEventStreamHandler(Type.publish, Event.msgsChanged, _updateMessages);
-        _messageListRepository.addListener(repositoryStreamHandler);
+        _repositoryStreamHandler = RepositoryEventStreamHandler(Type.publish, Event.msgsChanged, _updateMessages);
+        _messageListRepository.addListener(_repositoryStreamHandler);
       }
   }
 
   void _unregisterListeners() {
       if (_listenersRegistered) {
         _listenersRegistered = false;
-        _messageListRepository?.removeListener(repositoryStreamHandler);
+        _messageListRepository?.removeListener(_repositoryStreamHandler);
       }
   }
 
   void _updateMessages(Event event) => add(UpdateMessages());
 
   void _loadFlaggedMessages() async {
-    List<int> dateMakerIds = List();
-    Context context = Context();
-    List<int> messageIds = List.from(await context.getChatMessages(Chat.typeStarred, Context.chatListAddDayMarker));
+    final List<int> dateMakerIds = List();
+    final Context context = Context();
+    final List<int> messageIds = List.from(await context.getChatMessages(Chat.typeStarred, Context.chatListAddDayMarker));
+
+    if (null != _chatId) {
+      final List<int> messageIdsFromChat = List.from(await context.getChatMessages(_chatId, Context.chatListAddDayMarker));
+      messageIds.removeWhere((id) => !messageIdsFromChat.contains(id));
+      _messageListRepository.clear();
```

Comment:
I assume this is needed because otherwise all messages are returned after entering the "all flagged messages" view once, right?
I think we should alter the FlaggedMessagesLoaded response to not return everything that's located in the repository, but actually only the ids we gathered in `messageIds`. Then we don't have to clear the repository (which leads to reloading and dropping the cache for those items).

Reply:
That's correct. I changed it.